Nando Theessen

NandoTheessen opened the pull request:
## Newsfeed - jQuery

### Setup
- [x] Run less-watch-compiler and open index.html

### Part 1: The Menu
- [x] Import menu.less into index.less
- [x] Use jQuery to select the menu and menu button components
- [x] Write callback function to add/remove class to the menu
- [x] Look at LESS and determine what happens when classes are added/removed
- [x] Use jQuery to pass callback to a click event listener on the button

### Part 2: The Articles
- [x] Make sure the article LESS rules are being applied
- [x] Look over the LESS (but don't make changes yet)
- [x] Select all articles using jQuery and map over the array to create individual articles
- [x] In the Article class, set properties for element and expandButton
- [x] Add "click to expand" text to the button
- [x] Add click event to the expand button
- [x] Within the expand method, add or remove the open class from the article

### Part 3: Add Articles
- [x] Add your own articles to the HTML using the components we built

### Stretch Goals
- [x] Animate the menu opening
- [x] Animate the article opening
- [x] Add a close button to each Article component
- [x] Create a component constructor function
- [x] Implement a form with input fields to write new articles

## Good Habits
- [x] Pull Request title is your first and last name (add any notes in the comment field)
- [x] Code has been committed regularly
- [x] Commit comments are informative
- [x] Code is organized in a logical way

NandoTheessen committed:
Initial Commit

NandoTheessen committed:
Finished Part 1:

- Added a cb function w/ toggle
- Added on click event
- Used jQuery to grab elements from the Dom

NandoTheessen committed:
Finished Part 2 and 3

- Defined the class using the articles grabbed from the dom
- Filled the uttons with text
- Wrote cb function for our buttons
- implemented Article.js into html file
- created toggleClass function for articles

NandoTheessen committed:
Added animation for menu using the css

NandoTheessen committed:
Added animation for article opening

NandoTheessen committed:
Added close button

- Added functionality to close the article, but not reopen it

NandoTheessen committed:
Added Stretch goal component constructor:

- Start of the constructor function for new articles

JasonCorp84 commented on Article/Article.js:
```diff
@@ -36,4 +40,6 @@ class Article {
-articles = articles.map();
+articles = articles.map((index, element) => {
+  new Article(element);
+});
```
👍 👍 👍 
'Articles' 👿

JasonCorp84 commented on LESS/index.less:
```diff
@@ -5,6 +5,7 @@
 @import '../Header/Header.less';
 @import '../Menu/Menu.less';
+@import '../Article/Article.less';
 
```
👍👍👍

JasonCorp84 commented on index.html:
```diff
@@ -1,116 +1,126 @@
+  </div>
+</body>
+
 </html>
```
How about using JS + JQuery to feed news instead?

NandoTheessen replied:
Not 100% sure what you mean here!

NandoTheessen committed:
Finished the constructor function for new Articles:

- takes an Object as Argument
- Takes Title, Date, Body and Image out of that Object
- Creates a new Div with class .article and injects our content into that div
- Plan is to insert the input into html tags corresponding to it's natur (title -> h2 etc)
- Article retains all buttons and has an opening / closing function

NandoTheessen committed:
Multiple Changes to Article.js:

- Added comments to explain functionality

JasonCorp84 commented on Article/Article.js:
```diff
@@ -1,10 +1,11 @@
 let articles = $(".article");
+let buttons = $(".expandButton");
 
 
```
👍

JasonCorp84 commented on Article/Article.js:
```diff
@@ -33,6 +39,9 @@ class Article {
+  closeArticle() {
+    this.element.toggleClass("article-open");
+  }
 }
```
👍 👍 👍

NandoTheessen committed:
Added comments to explain functionality

JasonCorp84 commented on Article/Article.js:
```diff
@@ -55,6 +55,9 @@ articles = articles.map((index, element) => {
+// The new Article will use all existing css / and animations
 
 class NewArticle {
   constructor(articleObject) { // My Constructor will accept an Object and extracts Title, Date, Body (Text), and an Image
```
elegant solution 🥇

JasonCorp84 commented:
Nando,

Way to go Nando!
{3} <-- I know you will finish all the stretch assignments.

NandoTheessen committed:
Multiple changes:

- Added new file NewArticle.js housing the newArticle class
- Added input fields to grab new Article
- Added styles for input
- Implented new file into index.html

NandoTheessen committed:
Added functionality:

- Added funcitonality to submit button
- Added function to fill ArticleObject with user input
- Added functionality to create new Article using ArticleObject
- Added Expand / Close functionality to new Article (works for one new article)
